### ISSUE TITLE
Support hnix-store packages >= 0.5 

### ISSUE THREAD

**Issue by expipiplus1** [haskell-nix/hnix#952]
Looks like some functions have changed name and location.

Currently breaking in nixpkgs where only the latest hnix-store (and -remote) are available: https://github.com/NixOS/nixpkgs/pull/126693

@Anton-Latukha you seem to be behind the changes in hnix-store? Are you able to tackle this?

**Comment by sternenseemann** [comment]
If this is gonna take longer, downgrading the `hnix-store` packages in nixpkgs may also be a valid course of action.

**Comment by expipiplus1** [comment]
Indeed, hnix is the only non-broken package depending on them.

**Comment by Anton-Latukha** [comment]
Yes, I did `{Core,Remote}` changes & release, & guilty to forget to check Nixpkgs & submit a patch there.

Logically it is if one kinda needs an interpreter to work between the Nix store & the Nix configuration. Due to that `{Core,Remote}` are currently essentially an internal projects, they would get a lot more own traction transitively when HNix lib would work & be popular people would come over & if they would find they do not need HNix lib - they would use Store directly. Currently is Ok to freeze versions there: since https://packdeps.haskellers.com/reverse/hnix-store-core.

There is no problem supporting those changes in HNix, the culprit to make a release is #858, it is the last one to give GHC 9.0 downstream, & GHC 9.0 support is why I made `{Core,Remote}` migration breakages & releases in the first place.
